lib/ex_doc/formatter/html.ex

```diff
@@ -203,27 +203,42 @@ defmodule ExDoc.Formatter.HTML do
   end
 
   defp generate_extras(nodes_map, extras, config) do
-    extras
-    |> with_prev_next()
-    |> Enum.map(fn {node, prev, next} ->
-      filename = "#{node.id}.html"
-      output = "#{config.output}/#{filename}"
-      config = set_canonical_url(config, filename)
+    generated_extras =
+      extras
+      |> with_prev_next()
+      |> Enum.map(fn {node, prev, next} ->
+        filename = "#{node.id}.html"
+        output = "#{config.output}/#{filename}"
+        config = set_canonical_url(config, filename)
+
+        refs = %{
+          prev: prev && %{path: "#{prev.id}.html", title: prev.title},
+          next: next && %{path: "#{next.id}.html", title: next.title}
+        }
+
+        html = Templates.extra_template(config, node, nodes_map, refs)
+
+        if File.regular?(output) do
+          IO.puts(:stderr, "warning: file #{Path.relative_to_cwd(output)} already exists")
+        end
 
-      refs = %{
-        prev: prev && %{path: "#{prev.id}.html", title: prev.title},
-        next: next && %{path: "#{next.id}.html", title: next.title}
-      }
+        File.write!(output, html)
+        filename
+      end)
 
-      html = Templates.extra_template(config, node, nodes_map, refs)
+    generated_extras ++ copy_extras(config, extras)
+  end
 
-      if File.regular?(output) do
-        IO.puts(:stderr, "warning: file #{Path.relative_to_cwd(output)} already exists")
-      end
+  defp copy_extras(config, extras) do
+    for %{source_path: source_path} <- extras,
+        filename = Path.basename(source_path),
+        extension_name(filename) == ".livemd" do
+      output = "#{config.output}/#{filename}"
```

Comment:
I don't think we want the original filename, but we want the new .html file name. This is because we want to make it easy to find the .livemd from the HTML one. So I think this should be this:

```suggestion
    for %{source_path: source_path, id: id} <- extras,
        ext = extension_name(source_path),
        ext == ".livemd" do
      output = "#{config.output}/#{id}.#{ext}"
```

Maybe we should also have a test with a non-conventional .livemd filename?

Reply:
Makes sense, I'll get the tests added. For clarity, when you say "non-conventional" do you mean with spaces and special characters?

Reply:
Yes, exactly!